cpp/src/bindings.cpp

```diff
@@ -23,6 +22,55 @@ void bindArray(py::module_& m, const std::string& className) {
             }, py::keep_alive<0, 1>());
 }
 
+array* createArray(py::list columnsValues, py::list columnsNames, py::list columnsTypes, int columnSize, int columnsNumber) {
+    array* dataframe = new array();
+    dataframe->ptr = new series[columnsNumber];
+    series* columns = new series[columnsNumber];
+    for (int indice = 0 ; indice < columnsNumber ; indice ++ ) {
+        series* column = columns + indice;
+        py::str name = (py::str) columnsNames[indice];
+        column->name = strdup((char*) const_cast<char*>(((std::string) name).c_str()));
+        column->index = 0;
+        py::int_ type = (py::int_) columnsTypes[indice];
+        int properType = (int) type;
+        column->type = properType;
+        py::array data = (py::array) columnsValues[indice];
+        if (properType == 0) {
+            std::vector<std::string> values = py::cast<std::vector<std::string>>(columnsValues[indice]);
+            column->data.ptr = pypowsybl::copyVectorStringToCharPtrPtr(values);
+
+        } else if (properType == 1) {
+            std::vector<double> values = py::cast<std::vector<double>>(columnsValues[indice]);
+            column->data.ptr = pypowsybl::copyVectorDouble(values);
+        } else if (properType == 2 || properType == 3) {
+            std::vector<int> values = py::cast<std::vector<int>>(columnsValues[indice]);
+            column->data.ptr = pypowsybl::copyVectorInt(values);
+        }
+    }
+    dataframe->ptr = columns;
+    return dataframe;
+}
+
+void updateNetworkElementsWithSeries(pypowsybl::JavaHandle network, py::list columnsValues, py::list columnsNames, py::list columnsTypes, py::list indexList, int columnSize, int columnsNumber, element_type elementType, int indexCount) {
+    array* dataframe = createArray(columnsValues, columnsNames, columnsTypes, columnSize, columnsNumber);
+    char** charIndexList = pypowsybl::copyVectorStringToCharPtrPtr(py::cast<std::vector<std::string>>(indexList));
+    pypowsybl::updateNetworkElementsWithSeries(network, dataframe, charIndexList, columnSize, columnsNumber, elementType, indexCount);
+    pypowsybl::deleteCharPtrPtr(charIndexList, indexCount);
+    for (int indice = 0 ; indice < columnsNumber ; indice ++ ) {
+        series* columns = ((series*) dataframe->ptr) + indice;
+        if (columns->type == 0) {
+            pypowsybl::deleteCharPtrPtr((char**) columns->data.ptr, columnSize);
+        } else if (columns->type == 1) {
+            delete (double*) columns->data.ptr;
+        } else if (columns->type == 2) {
+            delete (int*) columns->data.ptr;
+        }
+        delete columns->name;
+    }
+    delete dataframe->ptr;
+    delete dataframe;
```

Comment:
We need to handle the case where the update will throw an exception, so that the .
For that the best option is probably to wrap the `array*` in a `std::shared_ptr<array>`  which will be destroyed when we leave the scope (exception or not).